smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/IncomingKafkaRecord.java

```diff
@@ -23,8 +28,18 @@ public IncomingKafkaRecord(
             KafkaCommitHandler commitHandler,
             KafkaFailureHandler onNack) {
         this.kafkaMetadata = new IncomingKafkaRecordMetadata<>(record);
-        this.metadata = Metadata.of(this.kafkaMetadata);
         this.commitHandler = commitHandler;
+
+        TracingMetadata tracingMetadata = TracingMetadata.empty();
```

Comment:
Should it be added only if tracing is enabled?

Reply:
Probably, will adjust